File: src/NodeParser/BooleanLiteralNodeParser.ts
import ts from "typescript";
import type { Context } from "../NodeParser.js";
import type { SubNodeParser } from "../SubNodeParser.js";
import type { BaseType } from "../Type/BaseType.js";
import { LiteralType } from "../Type/LiteralType.js";

export class BooleanLiteralNodeParser implements SubNodeParser {
    public supportsNode(node: ts.BooleanLiteral): boolean {
        return node.kind === ts.SyntaxKind.TrueKeyword || node.kind === ts.SyntaxKind.FalseKeyword;
    }
    public createType(node: ts.BooleanLiteral, context: Context): BaseType {
        return new LiteralType(node.kind === ts.SyntaxKind.TrueKeyword);
    }
}

File: internal/parser/boolean_literal_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// BooleanLiteralNodeParser handles `true` and `false` literals
// (src/NodeParser/BooleanLiteralNodeParser.ts).
type BooleanLiteralNodeParser struct{}

func NewBooleanLiteralNodeParser() *BooleanLiteralNodeParser {
	return &BooleanLiteralNodeParser{}
}

func (p *BooleanLiteralNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindTrueKeyword || node.Kind == ast.KindFalseKeyword
}

func (p *BooleanLiteralNodeParser) CreateType(node *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.LiteralType{Value: node.Kind == ast.KindTrueKeyword}
}
